src/collection-browser.ts

```diff
@@ -262,7 +262,7 @@ export class CollectionBrowser
   private dataSource: Record<string, TileModel[]> = {};
 
   @query('infinite-scroller')
-  private infiniteScroller!: InfiniteScroller;
+  private infiniteScroller?: InfiniteScroller;
```

Comment:
> This assertion is invalid in cases where the infinite scroller is not rendered, such as when showing a “0 results” or “no query” placeholder.

interesting... my mental model of collection browser does not make the infinitescroller optional, even without results or without a query started.
Q: what happens if we load empty infinite scroller when there aren't any results?

Reply:
Good question -- we could probably adjust the templates to keep the infinite scroller rendered to the DOM even when the placeholders replace it visually. Which, I concede, would remove the need for most of these changes 😄. Let me give that a try...

Reply:
hi, checking if an empty scroller worked?

Reply:
Thanks for the follow-up! I've updated the PR to keep an empty, hidden infinite scroller rendered even when the placeholders are shown. Seems to work fine, and should ensure that the infinite scroller always exists.